app/controllers/nodes_controller.rb

```diff
@@ -6,13 +6,17 @@ def show
 		else
 			@node = Node.root
 		end
-
 	end
 
 	def create
 		p = create_params
-		node = Node.create!(p)
-		redirect_to node_path(node.parent_id)
+		@node = Node.new(p)
+		if @node.save
+			flash[:success] = "Node created"
+		else
+			flash[:danger] = "Failed to create node"
+		end
+		redirect_to node_path(@node.parent_id)
```

Comment:
Update() has a different way of redirect_to depending on the possible problems while saving the node.  It's better if you keep the same convention (either redirect with error or always redirect to the parent like here)